### PR TITLE
Add some additional info to gRPC retry section

### PR DESCRIPTION
**Pull request by Sushisource**

### CODE REVIEW

File: docs/typescript/troubleshooting.md [hunk 1]
```diff
@@ -204,5 +204,12 @@ Error: 4 DEADLINE_EXCEEDED: context deadline exceeded
 ```
 
 This happens for a number of reasons: Network hiccups, timeouts that are too short, server overloaded.
+Querying a workflow whose query handler causes an error can result in the query call timing out.
```
Comment (flossypurse) [inline docs/typescript/troubleshooting.md:207]: Is this troubleshooting content specific to TS?

Comment (Sushisource) [reply docs/typescript/troubleshooting.md:207]: Nope, possible for anything

Comment (djmagee) [inline docs/typescript/troubleshooting.md:207]:
```suggestion
Querying a Workflow whose query handler causes an error can result in the query call timing out.
```

File: docs/typescript/troubleshooting.md [hunk 2]
```diff
@@ -204,5 +204,12 @@ Error: 4 DEADLINE_EXCEEDED: context deadline exceeded
 ```
 
 This happens for a number of reasons: Network hiccups, timeouts that are too short, server overloaded.
```
Comment (djmagee) [inline docs/typescript/troubleshooting.md:206]:
```suggestion
Several conditions can cause this error, including network hiccups, timeouts that are too short, and an overloaded server.
```

File: docs/typescript/troubleshooting.md [hunk 3]
```diff
@@ -204,5 +204,12 @@ Error: 4 DEADLINE_EXCEEDED: context deadline exceeded
...
+Querying a workflow whose query handler causes an error can result in the query call timing out.
 
-This is not your fault and likely indicative of an SDK bug. Please try to produce a minimal repro and we'll be glad to help.
+Some troubleshooting actions you can take:
+
+- Verify the connection from your worker to the Temporal Server is working and doesn't have unusually high latency
```
Comment (djmagee) [inline docs/typescript/troubleshooting.md:211]:
```suggestion
- Verify the connection from your Worker to the Temporal Server is working and doesn't have unusually high latency
```

File: docs/typescript/troubleshooting.md [hunk 4]
```diff
@@ -204,5 +204,12 @@ Error: 4 DEADLINE_EXCEEDED: context deadline exceeded
...
+- If you are running Temporal Server yourself, check your [server metrics](/docs/server/production-deployment/#scaling-and-metrics) to ensure it's not overloaded
+- If what's timing out is a query, check the logs of your workers to see if they are having issues handling the query
```
Comment (djmagee) [inline docs/typescript/troubleshooting.md:213]:
```suggestion
- If what's timing out is a query, check the logs of your Workers to see if they are having issues handling the query
```

File: docs/typescript/troubleshooting.md [hunk 5]
```diff
@@ -204,5 +204,12 @@ Error: 4 DEADLINE_EXCEEDED: context deadline exceeded
...
+
+If none of the above actions help you discover why timeouts are occuring, please try to produce a minimal repro and we'll be glad to help.
```
Comment (djmagee) [inline docs/typescript/troubleshooting.md:215]:
```suggestion
If none of the preceding actions help you discover why timeouts are occurring, please try to produce a minimal repro and we'll be glad to help.
```

File: docs/typescript/troubleshooting.md (hunk 3 above)
Comment (flossypurse) [reply docs/typescript/troubleshooting.md:211]: How does one verify this?

File: docs/typescript/troubleshooting.md (hunk 4 above)
Comment (flossypurse) [reply docs/typescript/troubleshooting.md:213]: How to check Worker logs ? 
Is this the same as looking at the logged output to the process - or is it writing to a file someplace?

File: docs/typescript/troubleshooting.md [hunk 6]
```diff
@@ -203,6 +203,13 @@ Error: 4 DEADLINE_EXCEEDED: context deadline exceeded
 }
 ```
 
-This happens for a number of reasons: Network hiccups, timeouts that are too short, server overloaded.
+Several conditions can cause this error, including network hiccups, timeouts that are too short, and an overloaded server.
+Querying a Workflow whose query handler causes an error can result in the query call timing out.
```
Comment (flossypurse) [inline docs/typescript/troubleshooting.md:207]:
```suggestion
Querying a Workflow Execution whose query handler causes an error can result in the query call timing out.
```

**Review by flossypurse — APPROVED** [review]
I would rather get this in now and figure out where to put it later.

**Pull request merged**